api/src/main/java/io/grpc/LoadBalancer.java

```diff
@@ -759,7 +760,8 @@ public boolean equals(Object other) {
       }
       CreateSubchannelArgs that = (CreateSubchannelArgs) other;
       return Objects.equal(addrs, that.addrs) && Objects.equal(attrs, that.attrs)
-          && Objects.equal(stateListener, that.stateListener);
+          && Objects.equal(stateListener, that.stateListener)
+          && Arrays.deepEquals(customOptions, that.customOptions);
```

Comment:
It's unclear what equality really means, but depending on the use-case the order that custom options were set should not matter, but here it does. (see also: hash code)

Reply:
Huh. Shouldn't this also apply for `EquivalentAddressGroup`?

Reply:
Order matters for sockets in EAG. It doesn't matter in Attributes. And in attributes we [ignore order](https://github.com/grpc/grpc-java/blob/v1.20.0/core/src/main/java/io/grpc/Attributes.java#L171).

It looks like we've not defined equals for CallOptions, not Context.